From f354cbaf93c13150ffb2b774d4c2e0737115a6d8 Mon Sep 17 00:00:00 2001
From: samiuddi <sami.uddin.mohammad@intel.com>
Date: Tue, 20 Feb 2018 10:28:54 +0530
Subject: [PATCH 2/2] Don't erase layer when we fail to find display after
 external display is unplugged.

Jira : NA
Test: we don't see any UI restart after unplug/plug
external display
Signed-off-by: samiuddi <sami.uddin.mohammad@intel.com>

Change-Id: Idb9d0ba091e32b93032f288f952fff787f32fd78
---
 graphics/composer/2.1/default/ComposerClient.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/graphics/composer/2.1/default/ComposerClient.cpp b/graphics/composer/2.1/default/ComposerClient.cpp
index e792034..0ab6c09 100644
--- a/graphics/composer/2.1/default/ComposerClient.cpp
+++ b/graphics/composer/2.1/default/ComposerClient.cpp
@@ -316,7 +316,12 @@ Return<Error> ComposerClient::destroyLayer(Display display, Layer layer)
         std::lock_guard<std::mutex> lock(mDisplayDataMutex);
 
         auto dpy = mDisplayData.find(display);
-        dpy->second.Layers.erase(layer);
+        if (dpy == mDisplayData.end()) {
+            ALOGE("failed to find display");
+            return Error::BAD_DISPLAY;
+        } else {
+            dpy->second.Layers.erase(layer);
+        }
     }
 
     return err;
-- 
1.9.1

